### PR TITLE
Refactor source and specs using modules

### PR DESCRIPTION
**Pull request by sindrenm**

### CODE REVIEW

**Comment by sindrenm** [comment]
Oh, and as @t0ggah noticed, I've changed the way we're testing the hyper-yield tag as well into what I believe adheres a little more to unit test practices, by isolating the hyper-yield tests from the hyper-content functionality. However, we should definitely keep the previous stuff as well as more of an integration test that ensures everything works as a whole!

**Comment by mathiasbno** [comment]
:+1:

**Comment by sindrenm** [comment]
:raised_hands:

**Comment by mathiasbno** [comment]
:gem: :package:

File: spec/directives/hyper-content.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  beforeEach(angular.mock.module('hyperContentFor'));
+
+  beforeEach(inject(function(_$compile_, _$rootScope_, _HYPER_CONTENT_FOR_IDS_) {
+    $compile = _$compile_;
+    $rootScope = _$rootScope_;
+    HYPER_CONTENT_FOR_IDS = _HYPER_CONTENT_FOR_IDS_;
+  }));
+
+  it('stores the transcluded content in the HYPER_CONTENT_FOR_IDS hash with the correct key', function() {
+    $compile("<hyper-content for='header'><h2>Sub-header</h2></hyper-yield>")($rootScope);
```
Comment (fespinoza) [inline spec/directives/hyper-content.spec.js:17]: `</hyper-yield>`?

File: spec/directives/hyper-content.spec.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    $rootScope.$digest();
+
+    expect(HYPER_CONTENT_FOR_IDS['header'][0].outerHTML).toEqual('<h2 class="ng-scope">Sub-header</h2>');
+  });
+
+  it('removes the transcluded content from the directive', function() {
+    var contentElement = $compile("<hyper-content for='header'><h2>Sub-header</h2></hyper-yield>")($rootScope);
```
Comment (fespinoza) [inline spec/directives/hyper-content.spec.js:24]: here too

File: spec/directives/hyper-content.spec.js (hunk 1 above)
Comment (sindrenm) [inline spec/directives/hyper-content.spec.js:17]: Oh! Thanks for noticing. I'll fix it on master. :smiley:

File: spec/directives/hyper-content.spec.js (hunk 2 above)
Comment (sindrenm) [inline spec/directives/hyper-content.spec.js:24]: Good eye!

**Pull request merged**